base/complex.jl

```diff
@@ -535,37 +535,42 @@ end
 #     return Complex(abs(iz)/r/2, copysign(r,iz))
 # end
 
-# compute exp(im*theta)
-function cis(theta::Real)
-    s, c = sincos(theta)
-    Complex(c, s)
-end
-
 """
-    cis(z)
+    cis(x)
 
-Return ``\\exp(iz)``.
+Computes a faster version of ``\\exp(i x)``, where `x` is a real or complex number.
+The `cis` name comes from the Euler formula of ``\\exp(i x) = cos(x) + i sin(x)`` where
+`cis` is shorthand for "cosine" plus "imaginary" times "sine".
```

Comment:
A vote to make this less verbose, while explaining why this it's a good idea:
```suggestion
Returns ``\\exp(i x) = \\cos(x) + i \\sin(x)``. 

This is usually faster than `exp(im*x)`, especially for real `x` where it can avoid 
calculating `sin` & `cos` of the real part of `im*x`.
```

Reply:
Yeah, I switched the order to make it more obvious.  Update coming.